frontend/public/components/dashboard/inventory-card/inventory-item.tsx

```diff
@@ -45,102 +54,161 @@ const getStatusGroupIcons = (flags: FlagsObject) => {
-    const title = count !== 1 ? pluralTitle : singularTitle;
-    let status: React.ReactNode;
-    if (error) {
-      status = <div className="co-dashboard-text--small text-secondary">Unavailable</div>;
-    } else if (isLoading) {
-      status = <LoadingInline />;
-    } else {
-      status = children;
-    }
-    return (
-      <div className="co-inventory-card__item">
-        <div className="co-inventory-card__item-title">{isLoading || error ? title : `${count} ${title}`}</div>
-        <div className="co-inventory-card__item-status">{status}</div>
+  ({ isLoading, title, count, children, error = false, TitleComponent, ExpandedComponent }) => {
+    const [expanded, setExpanded] = React.useState(false);
+    const pluralizedTitle = count !== 1 ? `${title}s` : title;
+    const titleMessage = isLoading || error ? pluralizedTitle : `${count} ${pluralizedTitle}`;
+    return ExpandedComponent ? (
+      <Accordion asDefinitionList={false} headingLevel="h5" className="co-inventory-card__accordion">
+        <AccordionItem>
+          <AccordionToggle
+            onClick={() => setExpanded(!expanded)}
+            isExpanded={expanded}
+            id={title}
+            className="co-inventory-card__accordion-toggle"
+          >
+            <div className="co-inventory-card__item">
+              <div className="co-inventory-card__item-title">
+                {(isLoading && !error) && <div className="skeleton-inventory" />}
+                {TitleComponent ? <TitleComponent>{titleMessage}</TitleComponent> : titleMessage}
+              </div>
+              {(!expanded && (error || !isLoading)) && (
+                <div className="co-inventory-card__item-status">
+                  {error ? <div className="co-dashboard-text--small text-secondary">Not available</div> : children}
+                </div>
+              )}
+            </div>
+          </AccordionToggle>
+          <AccordionContent isHidden={!expanded} className="co-inventory-card__accordion-body">
+            <ExpandedComponent />
+          </AccordionContent>
+        </AccordionItem>
+      </Accordion>
+    ) : (
+      <div className="co-inventory-card__item co-inventory-card__item--border">
+        <div className="co-inventory-card__item-title">
+          {(isLoading && !error) && <div className="skeleton-inventory" />}
+          {TitleComponent ? <TitleComponent>{titleMessage}</TitleComponent> : titleMessage}
+        </div>
+        {(error || !isLoading) && (
+          <div className="co-inventory-card__item-status">
+            {error ? <div className="co-dashboard-text--small text-secondary">Not available</div> : children}
+          </div>
+        )}
       </div>
     );
   }
 );
 
-export const Status: React.FC<StatusProps> = React.memo(({ groupID, count, flags }) => {
+export const Status = connectToFlags<StatusProps>(
+  ...getFlagsForExtensions(plugins.registry.getDashboardsInventoryItemGroups()),
+)(React.memo(({ groupID, count, flags }) => {
```

Comment:
Just curious, is the combo `connectToFlags` + `React.memo` efficient here? (we already do that in some other places)

When #2269 is done we should use e.g. `useExtensions` hook in combo with `React.memo`.

Reply:
It seems like there is no difference, so I removed `React.memo` when used with `connectToFlags`